### PR TITLE
Reading binaries/version for molecule tests from env variables

### PR DESCRIPTION
**Pull request by utkarsh5474**
# Description

https://confluentinc.atlassian.net/browse/ANSIENG-1047?atlOrigin=eyJpIjoiOWEyOWJkNmRlMzc4NDBiYTk3Nzc5YjM0NDk2MDc3MDYiLCJwIjoiaiJ9

For automating our patch releases, we need to read new binaries directly when running molecule tests. This change introduces env variables which'll resolve to new package binaries and new versions at runtime. 

With these change in place, to run a new patch, we'll need to set env variables COMMON_REPO_URL and VERSION before running the molecule tests to change dockerfiles. To make other relevant changes, we'll need to pass variables confluent_common_repository_baseurl and confluent_package_version/confluent_repo_version in molecule run. 

Default behaviour stays unchanged. 

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

cp-ansible-on-demand run 74. 

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

**Review by nsharma-git — COMMENTED** [review]
In a first look, seems you could have a variable for version. You don't need the lookup so many times.

**Comment by utkarsh5474** [comment]
Done @nsharma-git. Tested again with on-demand run 75.

File: roles/confluent.test/molecule/Dockerfile-debian.j2 [hunk 1]
```diff
@@ -12,26 +12,30 @@ RUN apt-get update && \
       unzip \
       curl
 
-RUN wget -qO - https://packages.confluent.io/deb/5.5/archive.key | sudo apt-key add - && \
-    add-apt-repository "deb [arch=amd64] https://packages.confluent.io/deb/5.5 stable main" && \
+{% set PACKAGE_VER = lookup('env', 'VERSION') | default('5.5.7', true) %}
+{% set REPO_VER = lookup('env', 'VERSION') | regex_replace('^([0-9])\\.([0-9]*).*', '\\1.\\2') | default('5.5', true) %}
```
Comment (nsharma-git) [inline roles/confluent.test/molecule/Dockerfile-debian.j2:16]: This is ok. A better way would be to use PACKAGE_VER. As you are sure at his variable holds the version. You may not need default case here

File: roles/confluent.test/molecule/Dockerfile-ubuntu.j2 [hunk 2]
```diff
@@ -26,26 +26,30 @@ RUN apt-get update -y && \
     && apt-get clean && \
     sed -i 's/^\($ModLoad imklog\)/#\1/' /etc/rsyslog.conf
 
-RUN wget -qO - https://packages.confluent.io/deb/5.5/archive.key | sudo apt-key add - && \
-    add-apt-repository "deb [arch=amd64] https://packages.confluent.io/deb/5.5 stable main" && \
+{% set PACKAGE_VER = lookup('env', 'VERSION') | default('5.5.7', true) %}
+{% set REPO_VER = lookup('env', 'VERSION') | regex_replace('^([0-9])\\.([0-9]*).*', '\\1.\\2') | default('5.5', true) %}
```
Comment (nsharma-git) [inline roles/confluent.test/molecule/Dockerfile-ubuntu.j2:30]: same as above

**Review by nsharma-git — COMMENTED** [review]
Looks good, Thanks!
A couple of nit pick.

File: roles/confluent.test/molecule/Dockerfile-debian.j2 (hunk 1 above)
Comment (utkarsh5474) [reply roles/confluent.test/molecule/Dockerfile-debian.j2:16]: Sorry @nsharma-git, I don't understand this. 
`VERSION` is an env variable that'll be set by the user (or by script reading extra vars in molecule run) 
PACKAGE_VER resolves it, by assuring that it has a default value. This default value will be diff for diff .x branches. 
REPO_VER picks out the x.y from x.y.z (VERSION) or defaults to 5.5 (which'll again be diff for diff .x branch)

Comment (utkarsh5474) [reply roles/confluent.test/molecule/Dockerfile-debian.j2:16]: Hi @nsharma-git, just bumping this.

Comment (utkarsh5474) [reply roles/confluent.test/molecule/Dockerfile-debian.j2:16]: Do you mean something like this? 
{% set DEFAULT_PACKAGE_VER = 5.5.7 %}
{% set DEFAULT_REPO_VER = DEFAULT_PACKAGE_VER | regex_replace('^([0-9])\\.([0-9]*).*', '\\1.\\2') %}
{% set PACKAGE_VER = lookup('env', 'VERSION') | default(DEFAULT_PACKAGE_VER, true) %}
{% set REPO_VER = lookup('env', 'VERSION') | regex_replace('^([0-9])\\.([0-9]*).*', '\\1.\\2') | default(DEFAULT_REPO_VER, true) %}

Comment (nsharma-git) [reply roles/confluent.test/molecule/Dockerfile-debian.j2:16]: I was thinking somethig like this - 
```
{% set PACKAGE_VER = lookup('env', 'VERSION') | default('5.5.7', true) %}
{% set REPO_VER =$PACKAGE_VER | regex_replace('^([0-9])\\.([0-9]*).*', '\\1.\\2')%}
```

**Review by nsharma-git — APPROVED** [review]

File: roles/confluent.test/molecule/Dockerfile-debian.j2 (hunk 1 above)
Comment (utkarsh5474) [reply roles/confluent.test/molecule/Dockerfile-debian.j2:16]: Yep, thanks. My bad. This is more elegant.

**Pull request merged**